modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C

```diff
@@ -28,56 +28,75 @@ CrossTermBarrierFunctionMaterial::CrossTermBarrierFunctionMaterial(const InputPa
     _prop_g(declareProperty<Real>(_function_name)),
     _prop_dg(_num_eta),
     _prop_d2g(_num_eta)
+
 {
+  //loop counters
+  unsigned int i,j;
+
   // if Vector W_ij is not the correct size to fill the matrix give error
   if (_num_eta * _num_eta != _W_ij.size())
     mooseError("Supply the number of etas squared for W_ij.");
 
   // if W_ij is not symmetric or if diagonal values are not zero, return error
-  for (unsigned int i = 0; i < _num_eta; ++i)
+  for (i = 0; i < _num_eta; ++i)
   {
     if (_W_ij[_num_eta * i + i] != 0)
       mooseError("Set on-diagonal values of W_ij to zero");
 
-    for (unsigned int j = i; j < _num_eta; ++j)
+    for (j = i; j < _num_eta; ++j)
     {
-      if (_W_ij[_num_eta * i + j] != _W_ij[_num_eta*j + i])
+      if (_W_ij[_num_eta * i + j] != _W_ij[_num_eta * j + i])
         mooseError("Supply symmetric values for W_ij");
     }
   }
 
+  for (i = 0; i < _num_eta; ++i)
+    _prop_d2g[i].resize(_num_eta);
+
   // declare derivative properties, fetch eta values
-  for (unsigned int i = 0; i < _num_eta; ++i)
+  std::vector<std::string> eta_name;
+  for (i = 0; i < _num_eta; ++i)
```

Comment:
This is your segfault right here.  `eta_name` has zero length, but you are accessing into it.